workflow/controller/operator.go

```diff
@@ -426,18 +431,23 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 		onExitNode, err = woc.executeTemplate(ctx, onExitNodeName, &wfv1.WorkflowStep{Template: exitHook.Template, TemplateRef: exitHook.TemplateRef}, tmplCtx, exitHook.Arguments, &executeTemplateOpts{onExitTemplate: true})
 		if err != nil {
 			x := fmt.Errorf("error in exit template execution : %w", err)
+			shouldDeletePVCs := true
+
 			switch err {
 			case ErrDeadlineExceeded:
 				woc.eventRecorder.Event(woc.wf, apiv1.EventTypeWarning, "WorkflowTimedOut", x.Error())
 			case ErrParallelismReached:
+				shouldDeletePVCs = false
 			default:
 				if !errorsutil.IsTransientErr(err) && !woc.wf.Status.Phase.Completed() && os.Getenv("BUBBLE_ENTRY_TEMPLATE_ERR") != "false" {
 					woc.markWorkflowError(ctx, x)
 				}
 			}
 			// Garbage collect PVCs if Onexit template execution returns error
-			if err := woc.deletePVCs(ctx); err != nil {
-				woc.log.WithError(err).Warn("failed to delete PVCs")
+			if shouldDeletePVCs {
+				if err := woc.deletePVCs(ctx); err != nil {
+					woc.log.WithError(err).Warn("failed to delete PVCs")
+				}
 			}
```

Comment:
All this logic appears to be a duplicate of the above, right? If we can do something about that it would be great.

Reply:
You're right 👍 I'll see what I can do..